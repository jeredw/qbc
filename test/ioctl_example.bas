'   Note: The IOCTL$ function works only if the device driver is
'         installed and states that it processes IOCTL strings, and
'         if BASIC performs an OPEN statement on the device. Do not
'         run this example in its current form.
'
OPEN "\DEV\ENGINE" FOR OUTPUT AS #1
IOCTL #1, "RAW"                   'Tells the device that the data is raw.
'
' If the character driver "ENGINE" responds "false" from
' the raw data mode in the IOCTL statement, then the file
' is closed.
'
IF IOCTL$(1) = "0" THEN CLOSE 1
