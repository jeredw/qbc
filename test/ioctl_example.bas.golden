(program statement \n statement \n statement \n statement \n statement \n 
  (statement 
    (open_statement OPEN 
      (expr 
        (literal "\DEV\ENGINE")) FOR 
      (open_mode OUTPUT) AS # 
      (expr 
        (literal 1)))) \n 
  (statement 
    (ioctl_statement IOCTL # 
      (expr 
        (literal 1)) , 
      (expr 
        (literal "RAW")))) \n statement \n statement \n statement \n statement \n statement \n 
  (statement 
    (if_inline_statement IF 
      (expr 
        (expr 
          (builtin_function IOCTL$ 
            ( 
              (expr 
                (literal 1)) ))) = 
        (expr 
          (literal "0"))) THEN 
      (if_inline_action 
        (statement 
          (close_statement CLOSE 
            (expr 
              (literal 1))))))) \n)
